Main/ClientApp/src/Actions/FaviconActions.ts

```diff
@@ -0,0 +1,30 @@
+import FaviconsApi from "../Api/FaviconsApi";
+import Dispatcher from "../Infrastructure/Dispatcher";
+import ActionTypes from "./ActionTypes";
+
+type GetFaviconsOkResponse = {
+  domain: string;
+  icons: [
+    {
+      src: string;
+      type?: string;
+      sizes?: string;
+    }
+  ];
+};
+
+export async function getFaviconUrl(domain: string) {
+  let response = await FaviconsApi.getFaviconUrl(domain);
+  if (response.ok) {
+    let faviconsResponse = (await response.json()) as GetFaviconsOkResponse;
+    if (faviconsResponse.icons[0] !== undefined) {
+      Dispatcher.dispatch({
```

Comment:
I really don't see any benefit of using Dispatcher/flux pattern here. Why not just await api call in the component's useEffect? 
Flux should be used for handling and storing state, using it for a single API call, which response is only used once is clearly overengineering and a lot of innecessary code